### PR TITLE
Implement JWT expiration time

### PR DESCRIPTION
**Pull request by vgeorge**
See #55. Changelog:

- Implement a default expiration time of 30 days for JWT session tokens
- Refactor configuration options, `APP_URL` env variable is dropped in favor of `PROTOCOL`, `HOST` and `PORT`
- Update of README

Test are passing, this is ready for a review.

### CODE REVIEW

File: test/test-auth.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        await client.get('/users');
+
+        // The test should never reach here, force execute catch block.
+        throw Error('An error was expected.');
+      } catch (error) {
+        // Check for the appropriate status response
+        expect(error.response.status).to.equal(401);
+      }
+    });
+    it('should refresh token near expiration date');
```
Comment (batpad) [inline test/test-auth.js]: Are these stub tests not yet written? Do we need to write them before merging?

File: README.md [hunk 2]
```diff
@@ -1,5 +1,14 @@
 # Observe API
 
+Data service for [Observe](https://github.com/developmentseed/observe) platform. This is a web server built in Node.js that provides an application programming interface for the [dashboard](https://github.com/developmentseed/observe-dashboard) and [mobile app](https://github.com/developmentseed/observe) of Observe.
+
+Documentation is available at:
+
+- [API](http://developmentseed.github.io/observe-api)
+- [Platform](https://github.com/developmentseed/observe#documentation) 
+
+This software is in early stages of development. Please visit the [issues page](https://github.com/developmentseed/observe-api/issues) to learn about known bugs and features development.
```
Comment (batpad) [inline README.md]: Very minor grammatical nitpick: `learn about known bugs and features development` -> `learn about known bugs and feature development`

**Comment by batpad** [comment]
@vgeorge this is looking really good to me.

I am getting some messages on the console when running the dev server that look a bit strange. Although I cannot confirm whether this is only due to the latest changes. Let me know if this is unexpected, and I can do some further digging to triage where this is coming from:

```
Server running at: http://0.0.0.0:3000
Debug: handler, error 
    Error: Not Found
    at internals.notFound (/home/sanj/seed/observe-api/node_modules/@hapi/hapi/lib/core.js:632:16)
    at module.exports.internals.Manager.execute (/home/sanj/seed/observe-api/node_modules/@hapi/hapi/lib/toolkit.js:41:33)
    at Object.internals.handler (/home/sanj/seed/observe-api/node_modules/@hapi/hapi/lib/handler.js:46:48)
    at exports.execute (/home/sanj/seed/observe-api/node_modules/@hapi/hapi/lib/handler.js:31:36)
    at Request._lifecycle (/home/sanj/seed/observe-api/node_modules/@hapi/hapi/lib/request.js:312:68)
    at process._tickCallback (internal/process/next_tick.js:68:7)
Debug: handler, error 
    Error: Not Found
    at internals.notFound (/home/sanj/seed/observe-api/node_modules/@hapi/hapi/lib/core.js:632:16)
    at module.exports.internals.Manager.execute (/home/sanj/seed/observe-api/node_modules/@hapi/hapi/lib/toolkit.js:41:33)
    at Object.internals.handler (/home/sanj/seed/observe-api/node_modules/@hapi/hapi/lib/handler.js:46:48)
    at exports.execute (/home/sanj/seed/observe-api/node_modules/@hapi/hapi/lib/handler.js:31:36)
    at Request._lifecycle (/home/sanj/seed/observe-api/node_modules/@hapi/hapi/lib/request.js:312:68)
    at process._tickCallback (internal/process/next_tick.js:68:7)

```

File: test/test-auth.js (hunk 1 above)
Comment (vgeorge) [reply test/test-auth.js]: We are not implemeting them now, removed.

**Comment by vgeorge** [comment]
@batpad I added fixes to address your feedback and did a force push to solve conflicts introduced by the PR #57. I'm not sure the console error you pointed is related to changes in this PR. The log line 2 (`Debug: handler, error`) should also includes `auth, jwt` if it was an authorization error. I noticed that sometimes Firefox open 2 or 3 OAuth popup windows, but I couldn't replicate the conditions this occurs. Please check the latest code. If tests are passing and you can login properly using the dashboard, I would merge this. We can open an issue about the error you are facing if we can isolate it better.

**Pull request merged**